http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/response/CrtResponseAdapter.java

```diff
@@ -127,8 +128,10 @@ private void onSuccessfulResponseComplete(HttpStream stream) {
 
     private void onFailedResponseComplete(HttpStream stream, HttpException error) {
         log.error(() -> "HTTP response encountered an error.", error);
-        responsePublisher.error(error);
-        failResponseHandlerAndFuture(stream, error);
+
+        IOException wrappedError = new IOException(error);
```

Comment:
Can we add a test to verify the behavior?

https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/aws-crt-client/src/test/java/software/amazon/awssdk/http/crt/AwsCrtHttpClientWireMockTest.java#L46

We can probably stub errors with something like this

https://github.com/aws/aws-sdk-java-v2/blob/639e57ccc612ea40e95cc8556943fe1fc333d39f/http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClientSpiVerificationTest.java#L72-L92